### PR TITLE
Record OpenSearch-Dashboards failed test cases in test-report.yml

### PR DESCRIPTION
**Pull request by peterzhuamazon**
### Description
Record OpenSearch-Dashboards failed test cases in test-report.yml

### Issues Resolved
https://github.com/opensearch-project/opensearch-metrics/issues/68

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/report_workflow/test_report_runner.py [hunk 1]
```diff
@@ -207,48 +215,63 @@ def get_os_cluster_logs(base_path: str, test_number: str, test_type: str, compon
     return [os_stdout, os_stderr]
 
 
-def get_failed_tests(base_path: str, test_number: str, test_type: str, component_name: str, config: str,
-                     product_name: str) -> typing.List[list]:
-    failed_test: list = []
-    result_path: str = ''
-    result_content: str = ''
+def get_failed_tests(product_name: str, test_result: str, test_result_files: list) -> typing.List[list]:
```
Comment (prudhvigodithi) [inline src/report_workflow/test_report_runner.py:218]: This method defaults to `.xml` and if OpenSearch will use the `html.parser` right ?

**Comment by peterzhuamazon** [comment]
> Hey @peterzhuamazon thanks, do we need all the `.xml` files to be part of the this change or can we just test on one `.xml`?

We need all of the xml, due to in cypress each plugin would run multiple test js file, and each xml corresponding to one js.

File: src/report_workflow/test_report_runner.py (hunk 1 above)
Comment (peterzhuamazon) [reply src/report_workflow/test_report_runner.py:218]: See code:
```
        if self.name == "opensearch":
            soup = BeautifulSoup(result_content, "html.parser")
        else:
            soup = BeautifulSoup(result_content, "xml")

```

**Review by prudhvigodithi — APPROVED** [review]

**Comment by prudhvigodithi** [comment]
Just a thought, I think for testing peter we can remove the all the `xml` files just keep one and update here https://github.com/opensearch-project/opensearch-build/pull/5111/files#diff-a44612e37c0352c6226189afb70d84c146f4dfe5982cfc6caf445e462a3664a3, then the tests should pass as well right?

**Comment by peterzhuamazon** [comment]
> Just a thought, I think for testing peter we can remove the all the `xml` files just keep one and update here https://github.com/opensearch-project/opensearch-build/pull/5111/files#diff-a44612e37c0352c6226189afb70d84c146f4dfe5982cfc6caf445e462a3664a3, then the tests should pass as well right?

We need to test the for loop with multiple xmls instead of only limit to one.

**Pull request merged**